beacon-chain/blockchain/service.go

```diff
@@ -89,14 +90,26 @@ type config struct {
 
 var ErrMissingClockSetter = errors.New("blockchain Service initialized without a startup.ClockSetter")
 
-type blobNotifierChan struct {
-	indices map[uint64]struct{}
-	channel chan struct{}
+type blobNotifierMap struct {
+	sync.RWMutex
+	notifiers map[[32]byte]chan uint64
 }
 
-type blobNotifier struct {
-	sync.RWMutex
-	chanForRoot map[[32]byte]*blobNotifierChan
+func (bn *blobNotifierMap) forRoot(root [32]byte) chan uint64 {
+	bn.Lock()
+	defer bn.Unlock()
+	c, ok := bn.notifiers[root]
+	if !ok {
+		c = make(chan uint64, fieldparams.MaxBlobsPerBlock)
+		bn.notifiers[root] = c
+	}
+	return c
+}
+
+func (bn *blobNotifierMap) delete(root [32]byte) {
+	bn.Lock()
+	defer bn.Unlock()
+	delete(bn.notifiers, root)
```

Comment:
If external code receives a channel from the `forRoot` method and then, while holding on to that channel, calls `delete` for the same root, there's a potential for deadlocks

Reply:
I don't understand the deadlock, can you explain it?

Reply:
Ignore. It's a non issue upon second look